mkinitcpio

```diff
@@ -251,6 +260,56 @@ build_image() {
     fi
 }
 
+build_uefi(){
+    set -x
+    local out=$1 microcode=$2 initramfs=$3 cmdline=$4 osrelease=$5 splash=$6 kernelimg=$7 uefistub=$8 errmsg=
+    OBJCOPYARGS=()
+
+    if [[ -z "$uefistub" ]]; then
+        for stub in {/usr,}/lib/systemd/boot/efi/linux{x64,ia32}.efi.stub {/usr,}/lib/gummiboot/linux{x64,ia32}.efi.stub; do
+            if [[ -f "$stub" ]]; then
+                uefistub="$stub"
+                break
+            fi
+        done
+    fi
+
+    if [[ -z "$kernelimg" ]]; then
+        for img in "/lib/modules/$KERNELIMAGE" "/boot/vmlinuz-$KERNELIMAGE" "/boot/vmlinuz-linuz"; do
+            if [[ -f "$img" ]]; then
+                KERNELIMAGE="$img"
+                break
+            fi
+        done
+    fi
+
+    if [[ -z "$cmdline" ]]; then
+        if [[ -f "/etc/kernel/cmdline" ]]; then
+            cmdline="/etc/kernel/cmdline"
+        else
+            cmdline="/proc/cmdline"
+        fi
+    fi
+
+    if [[ -n "$splash" ]]; then
+        OBJCOPYARGS+=("--add-section .splash=$splash --change-section-vma .splash=0x40000")
+    fi
+    msg "Creating EFI image: %s" "$out"
+    objcopy \
+        --add-section .osrel="$osrelease" --change-section-vma .osrel=0x20000 \
+        --add-section .cmdline="$cmdline" --change-section-vma .cmdline=0x30000 \
+        --add-section .linux="$kernelimg" --change-section-vma .linux=0x2000000 \
+        --add-section .initrd=<(cat "$microcode" "$initramfs" ) --change-section-vma .initrd=0x3000000 \
+        ${OBJCOPYARGS[@]} "$uefistub" "$out"
```

Comment:
Would it make sense to parse the $cmdline file, so that it could have embedded newlines and comments? The kernel command line can get a bit unwieldy, so my own scripts actually do something similar to:

```
--add-section .cmdline=<(grep '^[^#]' /etc/kernel/cmdline | tr -s '\n' ' ')
```

Admittedly this might result in a format that's incompatible with other existing implementations...

Reply:
This is the upstream code utilizing "/etc/kernel/cmdline" (i'm realizing I need to have better fallbacks to /usr/lib).

https://github.com/systemd/systemd/blob/183d5168cf1d2bfbd3049e8592b4ed4c87c94936/src/kernel-install/90-loaderentry.install#L65-L78

I think this would be a bad(?) idea and you are better off supplying this as your own option?

Reply:
the "upstream" `read -r -d '' -a BOOT_OPTIONS` does handle `\n` as a space

I'd personally prefer grawity's version where we can also add comments, but maybe for a future RFE?

Reply:
I went with grawity's version and waiting on feedback from grazzolini if it's okay or not :)